src/main.cpp

```diff
@@ -2134,6 +2226,14 @@ void UpdateCoins(const CTransaction& tx, CCoinsViewCache& inputs, CTxUndo &txund
                 undo.nVersion = coins->nVersion;
             }
         }
+
+        BOOST_FOREACH(const CTzeIn &tzein, tx.vtzein) {
+            CCoinsModifier coins = inputs.ModifyCoins(tzein.prevout.hash);
+            unsigned nPos = tzein.prevout.n;
+            coins->SpendTzeOut(nPos);
+
+            // TZE: Do we need a repr of undo information?
```

Comment:
Yes, we will need this. If a block containing a TZE input is orphaned, the spent TZE output needs to be re-added to the unspent TZE output set.